crates/core/machine/src/operations/field/field_op.rs

```diff
@@ -36,12 +36,67 @@ use typenum::Unsigned;
 pub struct FieldOpCols<T, P: FieldParameters> {
     /// The result of `a op b`, where a, b are field elements
     pub result: Limbs<T, P::Limbs>,
-    pub(crate) carry: Limbs<T, P::Limbs>,
+    pub carry: Limbs<T, P::Limbs>,
     pub(crate) witness_low: Limbs<T, P::Witness>,
     pub(crate) witness_high: Limbs<T, P::Witness>,
 }
 
 impl<F: PrimeField32, P: FieldParameters> FieldOpCols<F, P> {
+    #[allow(clippy::too_many_arguments)]
+    /// Populate result and carry columns from the equation (a*b + c) % modulus
+    pub fn populate_mul_and_carry(
+        &mut self,
+        record: &mut impl ByteRecord,
+        shard: u32,
+        a: &BigUint,
+        b: &BigUint,
+        c: &BigUint,
+        modulus: &BigUint,
+    ) -> (BigUint, BigUint) {
+        let p_a: Polynomial<F> = P::to_limbs_field::<F, _>(a).into();
+        let p_b: Polynomial<F> = P::to_limbs_field::<F, _>(b).into();
+        let p_c: Polynomial<F> = P::to_limbs_field::<F, _>(c).into();
+
+        let (result, carry) =
+            ((a * b + c) % modulus, (a * b + c - (a * b + c) % modulus) / modulus);
```

Comment:
Is it possible to make this clearer?

```suggestion
            let result = a * b + c;
            let (result, carry) = (result % modulus, (result - result % modulus) / modulus);
```